SU2_CFD/src/output/COutput.cpp

```diff
@@ -82,25 +82,28 @@ COutput::COutput(const CConfig *config, unsigned short ndim, bool fem_output):
   volumeFilename  = "volume";
   restartFilename = "restart";
 
-  /*--- Retrieve the history filename ---*/
 
-  historyFilename = config->GetConv_FileName();
 
   /*--- Add the correct file extension depending on the file format ---*/
 
   string hist_ext = ".csv";
   if (config->GetTabular_FileFormat() == TAB_OUTPUT::TAB_TECPLOT) hist_ext = ".dat";
 
-  /*--- Append the zone ID ---*/
+  /*--- Retrieve the history filename ---*/
+
+  historyFilename = config->GetConv_FileName();
```

Comment:
@pcarruscag so we add the hist_ext stuff to GetConv_Filename?

Reply:
Whenever a given file name (like the history or breakdown) is only used for a particular type of file, the accessor to get the file name can return it with extension.
The for output files whose extension depends on the type of format being output, we can add an accessor per file type, or make it clear in the name accessor that the name is without extension. However, it should contain everything else in the name that we need, zone idx, time step, etc.
I think this is the safest way to make this change while providing backward compatibility easily in one place.

Reply:
ok,
I move the entire filename construction into GetConv_FileName. 
I strip ".dat" and ".csv" only and keep other extensions.  Users can still use "history.dat", but using "history"  or "history.1.2.3.old" will now work.